packages/admin/admin/src/form/ChipSelect.tsx

```diff
@@ -0,0 +1,163 @@
+    `,
+);
+
+const InputBase = createComponentSlot(MuiInputBase)<ChipSelectClassKey>({
+    componentName: "ChipSelect",
+    slotName: "inputBase",
+})(
+    () => css`
+        .MuiInputBase-input {
+            padding: 0;
+            height: 100% !important;
+            min-height: unset !important;
+        }
+    `,
+);
+
+export interface ChipSelectProps<T = string>
+    extends ThemedComponentBaseProps<{
+            root: "div";
+            inputRoot: "div";
+            select: typeof Select;
+            menuItem: typeof MenuItem;
+            inputBase: typeof MuiInputBase;
+        }>,
+        Omit<ChipProps, "children" | "onChange"> {
+    children?: React.ReactNode;
+    getOptionLabel?: (option: T) => string;
+    getOptionValue?: (option: T) => string;
+    options?: T[];
+    selectedOption?: T;
+    onChange: (event: SelectChangeEvent) => void;
+    fullWidth?: boolean;
+}
+
+const ChipInput = ({
+    chipProps,
+    inputBaseProps,
+    inputRootProps,
+    ...restProps
+}: InputBaseProps & {
+    chipProps?: Omit<ChipProps, "children">;
+    inputBaseProps?: InputBaseProps;
+    inputRootProps?: React.ComponentPropsWithoutRef<"div">;
+}) => {
+    const fallBackLabel = <FormattedMessage id="comet.admin.form.chipselect.placeholder" defaultMessage="Select a value" />;
+
+    return (
+        <ChipInputRoot {...inputRootProps}>
+            <Chip
+                icon={<ChevronDown />}
+                label={restProps.value?.toString().length ? restProps.value : fallBackLabel}
+                variant="filled"
+                {...chipProps}
+            />
+            <InputBase
+                sx={{
+                    position: "absolute",
+                    top: 0,
+                    left: 0,
+                    right: 0,
+                    bottom: 0,
+                    opacity: 0,
+                }}
+                {...inputBaseProps}
+                {...restProps}
+            />
+        </ChipInputRoot>
+    );
+};
+
+export function ChipSelect<T = string>(inProps: ChipSelectProps<T>) {
+    const {
+        options,
+        getOptionLabel = (option: T) => {
+            if (typeof option === "object" && option !== null) {
+                if ((option as any).label) return String((option as any).label);
+                if ((option as any).id) return String((option as any).id);
+                if ((option as any).value) return String((option as any).value);
+                return JSON.stringify(option);
+            } else {
+                return String(option);
+            }
+        },
+        getOptionValue = (option: T) => {
+            if (typeof option === "object" && option !== null) {
+                if ((option as any).value) return String((option as any).value);
+                if ((option as any).id) return String((option as any).id);
+                return JSON.stringify(option);
+            } else {
+                return String(option);
+            }
+        },
```

Comment:
This seems a bit complicated and unsafe, when it comes to types... 
Why exactly is this needed?

Reply:
This implementation is alright. We cover most of our use cases with it, and therefore ease implementation for our colleagues.